scripts/specializations/CpCourseManager.lua

```diff
@@ -459,6 +463,13 @@ end
 function CpCourseManager:onUpdate()
     local spec = self.spec_cpCourseManager
     spec.courseRecorder:update()
+
+    --update lastWaypoint variable to keep moving the course when passing a new waypoint.
+    --only needs to be done when in show current waypoints mode to save performance
```

Comment:
Even better, don't call this on every update cycle as the current waypoint changes only few seconds.
Call for instance from `AIDriveStrategyFieldWorkCourse:onWaypointPassed`

Reply:
Thats smart. I havnt found that. But i will change it.